core/src/main/scala/kafka/server/KafkaRequestHandler.scala

```diff
@@ -108,6 +110,19 @@ class BrokerTopicMetrics(name: Option[String]) extends KafkaMetricsGroup {
   val failedFetchRequestRate = newMeter("FailedFetchRequestsPerSec", "requests", TimeUnit.SECONDS, tags)
   val totalProduceRequestRate = newMeter("TotalProduceRequestsPerSec", "requests", TimeUnit.SECONDS, tags)
   val totalFetchRequestRate = newMeter("TotalFetchRequestsPerSec", "requests", TimeUnit.SECONDS, tags)
+
+  val partitions = new ConcurrentHashMap[Int, Partition]
```

Comment:
Not sure I follow this. The case you're talking about is if `onPartitionAdd()` is invoked immediately after `onPartitionDelete()` with the same partition instance? It's not clear to me why this depends on the reference to `Partition`.